### PR TITLE
introduce phpstan

### PR DESCRIPTION
**Pull request by LordSimal**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -112,11 +112,15 @@ jobs:
...
 
     - name: Run phpcs
       run: vendor/bin/phpcs --report=checkstyle src/ tests/ | cs2pr
 
     - name: Run psalm
       if: success() || failure()
       run: vendor/bin/psalm.phar --output-format=github
+
+    - name: Run phpstan
+      if: success() || failure()
```
Comment (ADmad) [inline .github/workflows/ci.yml]:
```suggestion
      if: always()
```

File: composer.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "require-dev": {
         "cakephp/cakephp-codesniffer": "^4.0",
         "cakephp/authorization": "^2.0",
-        "phpunit/phpunit": "~8.5.0 | ^9.3"
+        "phpunit/phpunit": "~8.5.0 | ^9.3",
+        "cakedc/cakephp-phpstan": "^2.0"
```
Comment (ADmad) [inline composer.json]: Is this additional dependency really necessary?

File: src/Panel/DeprecationsPanel.php [hunk 3]
```diff
@@ -93,20 +93,9 @@ protected function _prepare()
...
         }
 
-        /** @psalm-suppress RedundantFunctionCall */
-        ksort($return['app']);
-        /** @psalm-suppress RedundantFunctionCall */
-        ksort($return['cake']);
-        /** @psalm-suppress RedundantFunctionCall */
         ksort($return['plugins']);
-        /** @psalm-suppress RedundantFunctionCall */
         ksort($return['vendor']);
```
Comment (markstory) [inline src/Panel/DeprecationsPanel.php:93]: Why are some sorts removed?

File: composer.json (hunk 2 above)
Comment (othercorey) [reply composer.json]: @LordSimal Do you have an update on this? Can we resolve this PR?

File: src/Panel/DeprecationsPanel.php (hunk 3 above)
Comment (ADmad) [reply src/Panel/DeprecationsPanel.php:93]: @markstory These array were lists, not associative arrays, so using key based sorting was redundant.

File: composer.json (hunk 2 above)
Comment (ADmad) [reply composer.json]: I have removed it.

**Pull request merged**